array.c

```diff
@@ -6291,7 +6291,7 @@ rb_ary_any_p(int argc, VALUE *argv, VALUE ary)
         }
     }
     else {
-	for (i = 0; i < RARRAY_LEN(ary); ++i) {
+	for (i = 0; i < len; ++i) {
 	    if (RTEST(rb_yield(RARRAY_AREF(ary, i)))) return Qtrue;
```

Comment:
I had thought the same thing in `Array#join` and then dug the history, so it appears that yield block may change the array length, so at least we cannot replace all `RARRAY_LEN(ary)` to `len`.

Reply:
Thanks for review.
I'll fix it.